app/assets/js/scripts/landing.js

```diff
@@ -445,35 +447,155 @@ const GAME_JOINED_REGEX = /\[.+\]: Sound engine started/
 const GAME_LAUNCH_REGEX = /^\[.+\]: (?:MinecraftForge .+ Initialized|ModLauncher .+ starting: .+|Loading Minecraft .+ with Fabric Loader .+)$/
 const MIN_LINGER = 5000
 
+// List of mods to exclude from validation
+const EXCLUDED_MODS = [
+]
+
 async function dlAsync(login = true) {
 
     // Login parameter is temporary for debug purposes. Allows testing the validation/downloads without
     // launching the game.
 
     const loggerLaunchSuite = LoggerUtil.getLogger('LaunchSuite')
-
+    const loggerLanding = LoggerUtil.getLogger('dlAsync')
     setLaunchDetails(Lang.queryJS('landing.dlAsync.loadingServerInfo'))
 
     let distro
 
     try {
         distro = await DistroAPI.refreshDistributionOrFallback()
         onDistroRefresh(distro)
-    } catch(err) {
-        loggerLaunchSuite.error('Unable to refresh distribution index.', err)
+    } catch (err) {
+        loggerLanding.error(Lang.queryJS('landing.dlAsync.unableToLoadDistributionIndex'))
         showLaunchFailure(Lang.queryJS('landing.dlAsync.fatalError'), Lang.queryJS('landing.dlAsync.unableToLoadDistributionIndex'))
         return
     }
 
     const serv = distro.getServerById(ConfigManager.getSelectedServer())
 
-    if(login) {
-        if(ConfigManager.getSelectedAccount() == null){
-            loggerLanding.error('You must be logged into an account.')
+    if (login) {
+        if (ConfigManager.getSelectedAccount() == null) {
+            loggerLanding.error(Lang.queryJS('landing.dlAsync.accountLoginNeeded'))
+            return
+        }
+    }
+
+    // --------- Mod Verification Logic ---------
+
+    if (extraFileVerif.status) {
+        loggerLanding.info(Lang.queryJS('landing.dlAsync.extraFileVerif.usingExtraFileVerif'))
+
+        const modsDir = path.join(ConfigManager.getDataDirectory(), 'instances', serv.rawServer.id, 'mods')
+
+        // Check if mods directory exists, if not, create it
+        if (!fs.existsSync(modsDir)) {
+            fs.mkdirSync(modsDir, {recursive: true})
+        }
+
+        const distroMods = {}
+        const mdls = serv.modules
+
+        // Populate expected mod identities and log them
+        mdls.forEach(mdl => {
+            if (mdl.rawModule.name.endsWith('.jar')) {
+                const modPath = path.join(modsDir, mdl.rawModule.name)
+                const modIdentity = mdl.rawModule || mdl.rawModule.artifact.MD5
```

Comment:
```suggestion
                const modIdentity = mdl.rawModule
```
MD5 can be blank, so I don't think it's a good idea to base the mod identity on that value. Also there will always be a rawModule property afaik

Reply:
I need to keep the .md5 file because if I don't include it, I won't be able to perform the verification. The MD5 is based on the file size, and during the creation of the distribution, it generates the hash. The value cannot be null.